var/spack/repos/builtin/packages/cuda/package.py

```diff
@@ -702,6 +702,47 @@ class Cuda(Package):
     provides("opencl@:1.2", when="@7:")
     provides("opencl@:1.1", when="@:6")
 
+    for cuda_ver, cub_ver in [
+        ["12.6.3", "2.5.0"],
+        ["12.6.2", "2.5.0"],
+        ["12.6.1", "2.5.0"],
+        ["12.6.0", "2.5.0"],
+        ["12.5.1", "2.4.0"],
+        ["12.5.0", "2.4.0"],
+        ["12.4.1", "2.3.1"],
+        ["12.4.0", "2.3.1"],
+        ["12.3.2", "2.2.0"],
+        ["12.3.1", "2.2.0"],
+        ["12.3.0", "2.2.0"],
+        ["12.2.2", "2.1.0"],
+        ["12.2.1", "2.1.0"],
+        ["12.2.0", "2.0.1"],
+        ["12.1.1", "2.0.1"],
+        ["12.1.0", "2.0.1"],
+        ["12.0.1", "2.0.1"],
+        ["12.0.0", "2.0.1"],
+        ["11.8.0", "1.15.1"],
+        ["11.7.1", "1.15.0"],
+        ["11.7.0", "1.15.0"],
+        ["11.6.2", "1.15.0"],
+        ["11.6.1", "1.15.0"],
+        ["11.6.0", "1.15.0"],
+        ["11.5.2", "1.13.1"],
+        ["11.5.1", "1.13.1"],
+        ["11.5.0", "1.13.1"],
+        ["11.4.4", "1.12.1"],
+        ["11.4.3", "1.12.1"],
+        ["11.4.2", "1.12.1"],
+        ["11.4.1", "1.12.1"],
+        ["11.4.0", "1.12.1"],
+        ["11.3.1", "1.11.0"],
+        ["11.3.0", "1.11.0"],
+        ["11.2.2", "1.10.0"],
+        ["11.2.1", "1.10.0"],
+        ["11.2.0", "1.10.0"],
+    ]:
```

Comment:
can this be simplified to

```python
provides("cub-api@1.10.0", when="@11.2")
provides("cub-api@1.11.0", when="@11.3")
...
```
and preferably in a declarative way so it can be grepped instead of imperative with string interpolation etc?